include/NAS2D/MathUtils.h

```diff
@@ -29,5 +30,37 @@ namespace NAS2D
 
 	int divideUp(int to_divide, int divisor);
 
+	
+
+/**
+ * \fn OutputType scaleLinear(const InputType& a, const OutputType& b)
+ *
+ * Applies a linear conversion to map an input domain to an output range.
+ *
+ * \param	value: The value to be converted from the input domain to the output range.
+ * \param	domainPoint1: A known lower fixed point in the domain. Must be less than domainPoint2.
+ * \param	domainPoint2: A known upper fixed point in the domain. Must be greater than domainPoint1.
+ * \param	rangePoint1: A known lower fixed point in the range. Must be less than or equal to rangePoint2.
+ * \param	rangePoint2: A known upper fixed point in the range. Must be greater than or equal to rangePoint1.
+ * \return	Returns value as mapped to the output range.
+ * \remarks
+ * `value` is not required to start within [domainPoint1,domainPoint2]. E.g. temperature conversions:
+ *
+ *     scaleLinear(0.0f, 32.0f, 212.0f, 0.0f, 100.0f)
+ *
+ * will convert 0 degees Fahrenheit to its corresponding value `-17.77777` degrees Celsius.
+ * 
+ * Avoid inputs such that the difference of domainPoint2 and domainPoint1 is zero. Otherwise, a divide by zero error could occur.
+ * 
+ * If the difference of rangePoint2 and rangePoint1 is zero. The result of the function is always rangePoint1.
```

Comment:
I was kind of thinking this might be simpler to talk about points between equal, rather than having a difference of zero. I suppose this is technically more accurate though, since you can create an exotic type where they have a distance of zero but are not equal. Just wanted to draw attention to that technicality.